### PR TITLE
jailer: Ensure that bindmounts are propagated into the root

### PR DESCRIPTION
**Pull request by mcastelino**
User can bind mount into the chroot location.
This is needed as hard links cannot cross file system boundaries.
Copy is not always feasible (e.g. block devices).

Change the bind mount to be slave, such that host to jail bind
mounts are properly propagated. However we do not want to jail
to host events to propgate back.

Fixes: #1089

Signed-off-by: Manohar Castelino <manohar.r.castelino@intel.com>

Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jailer/src/chroot.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ pub fn chroot(path: &Path) -> Result<()> {
             null(),
             root_dir.as_ptr(),
             null(),
-            libc::MS_PRIVATE | libc::MS_REC,
+            libc::MS_SLAVE | libc::MS_REC,
```
Comment (alexandruag) [inline jailer/src/chroot.rs]: Would be nice to also change `PRIVATE` to `SLAVE` in the comment on line 30 (it's not the most useful comment; basically `pivot_root` did no work with `MS_SHARED` mount points a while ago; dunno if anything changed), and rename the `Error::MountPropagationPrivate` enum variant (used on line 42) to `MountPropagationSlave`.

Comment (mcastelino) [reply jailer/src/chroot.rs]: Yes. I missed that. Also the error reported needs to be changed too. Will fix both.

Comment (mcastelino) [reply jailer/src/chroot.rs]: Fixed and updated the PR

Comment (petreeftime) [reply jailer/src/chroot.rs]: The comment above still says `// Recursively change the propagation type of all the mounts in this namespace to **PRIVATE**, so`

**Pull request merged**